### PR TITLE
Poc/relational data with refactoring

### PR DESCRIPTION
**Pull request by sl-slaing**
Demonstration of changes in the PoC

### CODE REVIEW

File: core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/ExtentAugmentedFields.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package com.scottlogic.datahelix.generator.core.generation.relationships;
+
+import com.scottlogic.datahelix.generator.common.profile.Field;
+import com.scottlogic.datahelix.generator.common.profile.FieldType;
+import com.scottlogic.datahelix.generator.common.profile.Fields;
+import com.scottlogic.datahelix.generator.common.profile.SpecificFieldType;
+
+public class ExtentAugmentedFields extends Fields {
```
Comment (sl-slaing) [inline core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/ExtentAugmentedFields.java]: Would be better if there was an interface for this type, with `getMin` and `getMax` methods

File: core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/ExtentAugmentedFields.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    public static final Field min = new Field("min", integer, false, null, false, true, null);
+    public static final Field max = new Field("max", integer, false, null, false, true, null);
+
+    public ExtentAugmentedFields(Fields fields) {
+        super(fields.asList());
+    }
+
+    @Override
+    public Field getByName(String fieldName) {
```
Comment (sl-slaing) [inline core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/ExtentAugmentedFields.java]: Is this required?

File: core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/OneToManyRangeResolver.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        return range;
+    }
+
+    private static OneToManyRange applyAtomicConstraint(OneToManyRange range, AtomicConstraint atomicConstraint) {
+        Field field = atomicConstraint.getField();
+
+        int newExtent = getExtent(atomicConstraint);
+
+        if (field.equals(min)) {
+            return range.withMin(newExtent);
```
Comment (sl-slaing) [inline core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/OneToManyRangeResolver.java]:
```suggestion
            return range.withMin(getExtent(atomicConstraint));
```

File: core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/OneToManyRangeResolver.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        } else if (field.equals(max)) {
+            return range.withMax(newExtent);
```
Comment (sl-slaing) [inline core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/OneToManyRangeResolver.java]:
```suggestion
            return range.withMax(getExtent(atomicConstraint));
```

File: core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/OneToManyRangeResolver.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+                OneToManyRangeResolver::applyAtomicConstraint,
+                (a, b) -> null);
+
+        return range;
+    }
+
+    private static OneToManyRange applyAtomicConstraint(OneToManyRange range, AtomicConstraint atomicConstraint) {
+        Field field = atomicConstraint.getField();
+
+        int newExtent = getExtent(atomicConstraint);
```
Comment (sl-slaing) [inline core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/OneToManyRangeResolver.java]:
```suggestion
```

File: core/src/main/java/com/scottlogic/datahelix/generator/core/guice/DataGeneratorProvider.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        @Named("config:maxRows") long maxRows,
+        DataGeneratorMonitor monitor) {
+        this.coreGenerator = coreGenerator;
+        this.maxRows = maxRows;
+        this.monitor = monitor;
+    }
+
+    @Override
+    public DataGenerator get() {
+        return new MonitoringDataGenerator(
```
Comment (sl-slaing) [inline core/src/main/java/com/scottlogic/datahelix/generator/core/guice/DataGeneratorProvider.java]: This should only be included if `--quiet` isn't supplied?

File: examples/faker/relational.json [hunk 7]
```diff
@@ -0,0 +1,110 @@
+{
```
Comment (sl-slaing) [inline examples/faker/relational.json]: Should be removed

File: core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/OneToManyRangeResolver.java (hunk 3 above)
Comment (laingsimon) [reply core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/OneToManyRangeResolver.java]: Done

File: core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/OneToManyRangeResolver.java (hunk 4 above)
Comment (laingsimon) [reply core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/OneToManyRangeResolver.java]: Done

File: core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/OneToManyRangeResolver.java (hunk 5 above)
Comment (laingsimon) [reply core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/OneToManyRangeResolver.java]: Done

File: examples/faker/relational.json (hunk 7 above)
Comment (laingsimon) [reply examples/faker/relational.json]: Done

File: core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/ExtentAugmentedFields.java (hunk 2 above)
Comment (laingsimon) [reply core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/ExtentAugmentedFields.java]: Used by `ConstraintService.createConstraint:130`

File: core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/ExtentAugmentedFields.java (hunk 1 above)
Comment (laingsimon) [reply core/src/main/java/com/scottlogic/datahelix/generator/core/generation/relationships/ExtentAugmentedFields.java]: Refactored

**Pull request closed without merging**